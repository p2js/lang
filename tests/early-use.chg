mod foo {
    fn bar(): i32 { return 42; }
    mod nested {
        fn baz() {}
    }
}
use foo::bar;
use foo::nested;
fn main(): i32 {
    nested::baz();
    return bar();
}
