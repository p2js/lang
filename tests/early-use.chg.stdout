expr @ 2:28 2:30 => i32
expr @ 2:21 2:30 => never
expr @ 2:19 3:0 => ()
expr @ 6:12 6:15 => fn bar(...)
expr @ 6:12 6:17 => i32
expr @ 6:5 6:17 => never
expr @ 5:16 8:1 => ()



define i32 @root$foo$bar() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 42, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}

define i32 @main() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
%l.1 = call i32 @root$foo$bar()
br label %bb.1

bb.1:
store i32 %l.1, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}


